app/src/main/java/de/danoeh/antennapod/dialog/TagSettingsDialog.java

```diff
@@ -27,28 +27,49 @@
 import io.reactivex.schedulers.Schedulers;
 
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class TagSettingsDialog extends DialogFragment {
     public static final String TAG = "TagSettingsDialog";
     private static final String ARG_FEED_PREFERENCES = "feed_preferences";
+    private static final String ARG_MULTI_FEED_PREFERENCES = "multi_feed_preferences";
     private List<String> displayedTags;
     private EditTagsDialogBinding viewBinding;
     private TagSelectionAdapter adapter;
 
-    public static TagSettingsDialog newInstance(FeedPreferences preferences) {
+    private static TagSettingsDialog newInstance() {
         TagSettingsDialog fragment = new TagSettingsDialog();
         Bundle args = new Bundle();
-        args.putSerializable(ARG_FEED_PREFERENCES, preferences);
+        args.putSerializable(ARG_FEED_PREFERENCES, null);
         fragment.setArguments(args);
         return fragment;
     }
 
+    public static TagSettingsDialog newInstance(FeedPreferences preferences) {
```

Comment:
I would just call `newInstance(Collections.singletonList(preferences))` here to reduce code complexity. Then you don't need any case distinctions between single feeds and multiple feeds. Then you can also remove the no-argument `newInstance` method.